toolkit/scripts/toolchain.mk

```
@@ -169,10 +171,25 @@ $(final_toolchain): $(raw_toolchain) $(STATUS_FLAGS_DIR)/build_toolchain_srpms.f
 .SILENT: $(toolchain_rpms)
 
 ifeq ($(REBUILD_TOOLCHAIN),y)
+
+# Our manifest files should always track the contents of the freshly built archives exactly
+#   Currently non-blocking, to make this a blocking check switch to `print_error` instead of
+#   `print_warning`
+$(STATUS_FLAGS_DIR)/toolchain_verify.flag: $(TOOLCHAIN_MANIFEST) $(final_toolchain)
+	@echo Validating contents of toolchain against manifest...
+	tar -tzf $(final_toolchain) | sed 's+built_rpms_all/++g' | sed '/^$$/d' | sort > $(toolchain_actual_conents)
+	sort $(TOOLCHAIN_MANIFEST) > $(toolchain_expected_conents)
+	@diff="$$(comm -3 $(toolchain_actual_conents)  $(toolchain_expected_conents) )" && \
+	if [ -n "$${diff}" ]; then \
+		echo "ERROR: Missmatched packages between '$(TOOLCHAIN_MANIFEST)' and '$(final_toolchain)':" && \
+		echo "$${diff}"; \
```

For the sake of consistency:

```suggestion
	@echo Validating contents of toolchain against manifest... && \
	tar -tzf $(final_toolchain) | sed 's+built_rpms_all/++g' | sed '/^$$/d' | sort > $(toolchain_actual_conents) && \
	sort $(TOOLCHAIN_MANIFEST) > $(toolchain_expected_conents) && \
	@diff="$$(comm -3 $(toolchain_actual_conents)  $(toolchain_expected_conents) )" && \
	if [ -n "$${diff}" ]; then \
		echo "ERROR: Missmatched packages between '$(TOOLCHAIN_MANIFEST)' and '$(final_toolchain)':" && \
		echo "$${diff}"; \
```

No entirely sure, if I wrote this correctly, though.:)

---

Looks good. I had those split out for debugging purposes (so it would print), but forgot to add them back. The `@` before the diff= needs to be removed since the one on the first line will handle it. I'll update on my end.